module/VuFindTheme/src/VuFindTheme/View/Helper/AssetManager.php

```diff
@@ -0,0 +1,402 @@
+     * @param array  $attributes Extra attributes for style tag
+     *
+     * @return void
+     */
+    public function appendStyle(string $css, array $attributes = []): void
+    {
+        $this->styles[] = compact('css', 'attributes');
+    }
+
+    /**
+     * Add an entry to the list of stylesheets.
+     *
+     * @param string $href                  Stylesheet href
+     * @param string $media                 Media
+     * @param string $conditionalStylesheet Any conditions
+     * @param array  $extras                Array of extra attributes
+     *
+     * @return void
+     */
+    public function appendStylesheet(
+        string $href,
+        string $media = 'screen',
+        string $conditionalStylesheet = '',
+        array $extras = []
+    ): void {
+        $this->stylesheets[] = compact('href', 'media', 'conditionalStylesheet', 'extras');
+    }
+
+    /**
+     * Forcibly prepend a stylesheet, removing it from any existing position
+     *
+     * @param string $href                  Stylesheet href
+     * @param string $media                 Media
+     * @param string $conditionalStylesheet Any conditions
+     * @param array  $extras                Array of extra attributes
+     *
+     * @return void
+     */
+    public function forcePrependStylesheet(
+        string $href,
+        string $media = 'screen',
+        string $conditionalStylesheet = '',
+        array $extras = []
+    ): void {
+        $newSheets = [compact('href', 'media', 'conditionalStylesheet', 'extras')];
+        foreach ($this->stylesheets as $sheet) {
+            if ($sheet['href'] !== $newSheets[0]['href']) {
+                $newSheets[] = $sheet;
+            }
+        }
+        $this->stylesheets = $newSheets;
+    }
+
+    /**
+     * Clear the list of stylesheets, re-establishing it with the provided one.
+     *
+     * @param string $href                  Stylesheet href
+     * @param string $media                 Media
+     * @param string $conditionalStylesheet Any conditions
+     * @param array  $extras                Array of extra attributes
+     *
+     * @return void
+     */
+    public function setStylesheet(
+        string $href,
+        string $media = 'screen',
+        string $conditionalStylesheet = '',
+        array $extras = []
+    ): void {
+        $this->stylesheets = [compact('href', 'media', 'conditionalStylesheet', 'extras')];
+    }
+
+    /**
+     * Append raw script code.
+     *
+     * @param string $script              Script code
+     * @param string $type                Script type
+     * @param array  $attrs               Additional attributes for the script tag
+     * @param bool   $allowArbitraryAttrs Should we allow arbitrary attributes in $attrs?
+     * @param string $position            Position to output script (header or footer)
+     *
+     * @return void
+     */
+    public function appendScript(
+        string $script,
+        string $type = 'text/javascript',
```

Comment:
Since the [type attribute should not be added for JS](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type#attribute_is_not_set_default_an_empty_string_or_a_javascript_mime_type) anymore, perhaps this should not be a parameter at all in script methods. If there's any other type of script, it could be added to the attributes.

Reply:
My VM crashed while I was testing it and I'm now out of time to try again, but I'm pretty sure that 6a2df6b79e42aceeeb5d2249db552afb15a27dfc should take care of this.

Reply:
...and I finally managed to re-run tests today, and can confirm that they're still passing.